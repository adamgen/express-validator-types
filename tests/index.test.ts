import { describe, it, expect } from '@jest/globals';
import {
  testFileTypeKeyValues,
  getProjectFile,
} from '../jest/testFileTypeKeyValues';

describe('schema to ts', () => {
  it('shuold convert', () => {
    expect(
      testFileTypeKeyValues(
        [__dirname, 'tsconfig.json'],
        'index.ts',
        'myObject'
      )
    ).toEqual({
      name: 'string',
      id: 'number',
      resource_tag_ids: 'number[]',
      rsa_private: 'unknown',
      rsa_public: 'unknown',
    });

    const indexTs = getProjectFile([__dirname, 'tsconfig.json'], 'index.ts');
    const schemaConst = indexTs.getVariableDeclarationOrThrow('schema');
    const schemaConstType = schemaConst.getType();
    const schemaConstProps = schemaConstType.getProperties();

    const actual = {};
    for (const prop of schemaConstProps) {
      actual[prop.getFullyQualifiedName()] = prop
        .getTypeAtLocation(schemaConst)
        .getText();
    }
    expect(actual).toMatchSnapshot();
  });
});
