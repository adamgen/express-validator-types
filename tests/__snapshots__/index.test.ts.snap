// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema to ts shuold convert 1`] = `
{
  "__object.'resource_tag_ids.*'": "{ readonly in: "body"; readonly isInt: true; readonly optional: true; }",
  "__object.id": "{ readonly in: "body"; readonly isInt: true; readonly exists: { readonly negated: true; }; }",
  "__object.name": "{ readonly in: "body"; readonly isString: true; readonly isEmpty: { readonly negated: true; }; readonly optional: true; }",
  "__object.resource_tag_ids": "{ readonly in: "body"; readonly isArray: true; readonly optional: true; }",
  "__object.rsa_private": "{ readonly in: "body"; readonly exists: { readonly negated: true; }; }",
  "__object.rsa_public": "{ readonly in: "body"; readonly custom: { readonly options: (rsa_public: any) => boolean; }; }",
}
`;
